cranelift/codegen/src/isa/x64/inst/mod.rs

```diff
@@ -2211,6 +2211,17 @@ impl MachInst for Inst {
         Inst::nop(std::cmp::min(preferred_size, 15) as u8)
     }
 
+    fn align_basic_block(offset: CodeOffset, loop_header: bool) -> CodeOffset {
+        if loop_header {
+            // Unaligned loop headers can cause severe performance problems.
+            // See https://github.com/bytecodealliance/wasmtime/issues/4883.
+            // Here we use conservative 64 bytes alignment.
+            align_to(offset, 64)
```

Comment:
Could you disable this when optimizing for size? Also won't this cause a large blowup when there are a lot of tiny loops in a function with multiple being able to fit in a single cacheline?

Reply:
Also maybe disable it for cold blocks?